web-app-router/components/Table/index.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from "react";
+import { CaretIcon } from "../Icons/CaretIcon";
+import {
+  Select,
+  SelectButton,
+  SelectOption,
+  SelectOptions,
+} from "@/components/Select";
+import clsx from "clsx";
+
+type TableProps = {
+  headers: (React.ReactNode | null | undefined)[];
+  rows: React.ReactNode[][];
+  totalResults: number;
+  rowsPerPageOptions: number[];
+  initialRowsPerPage?: number;
+  onPageChange: (page: number) => void;
+  onRowsPerPageChange: (rowsPerPage: number) => void;
+  currentPage: number;
+};
+
+export const TableComponent: React.FC<TableProps> = ({
+  headers,
+  rows,
+  totalResults,
+  rowsPerPageOptions,
+  initialRowsPerPage = rowsPerPageOptions[0],
+  onPageChange,
+  onRowsPerPageChange,
+  currentPage,
+}) => {
+  const [rowsPerPage, setRowsPerPage] = useState(initialRowsPerPage);
+
+  const handleRowsPerPageChange = (newRowsPerPage: number) => {
+    setRowsPerPage(newRowsPerPage);
+    onRowsPerPageChange(newRowsPerPage);
+  };
+
+  const handlePageChange = (newPage: number) => {
+    onPageChange(newPage);
+  };
+
+  // Calculate the number of pages
+  const pageCount = Math.ceil(totalResults / rowsPerPage);
+
+  return (
+    <div className="w-full h-full">
+      <table className="min-w-full divide-y divide-gray-100">
```

Comment:
@andy-t-wang Also, we need to remember to use `grey` instead of `gray`. gray-100 is a different color from the tailwind default config. Please, fix it everywhere.